verticapy/tests_new/machine_learning/vertica/model_utils.py

```diff
@@ -0,0 +1,1558 @@
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless  required  by applicable  law or  agreed to in
+writing, software  distributed  under the  License is
+distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, either express or implied.
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+from itertools import chain
+import sys
+
+import numpy as np
+import pytest
+import sklearn.cluster as skl_cluster
+import sklearn.ensemble as skl_ensemble
+import sklearn.linear_model as skl_linear_model
+import sklearn.tree as skl_tree
+import sklearn.dummy as skl_dummy
+from sklearn.preprocessing import LabelEncoder
+import sklearn.svm as skl_svm
+from statsmodels.tsa.arima.model import ARIMA
+import xgboost as xgb
+
+import verticapy as vp
+from verticapy.connection import current_cursor
+import verticapy.machine_learning.vertica as vpy_linear_model
+import verticapy.machine_learning.vertica.cluster as vpy_cluster
+import verticapy.machine_learning.vertica.ensemble as vpy_ensemble
+import verticapy.machine_learning.vertica.svm as vpy_svm
+import verticapy.machine_learning.vertica.tree as vpy_tree
+import verticapy.machine_learning.vertica.tsa as vpy_tsa
+from verticapy.tests_new.machine_learning.vertica import (
+    REGRESSION_MODELS,
+    CLASSIFICATION_MODELS,
+    TIMESERIES_MODELS,
+    CLUSTER_MODELS,
+)
+
+le = LabelEncoder()
+
+if sys.version_info < (3, 12):
+    import tensorflow as tf
+
+
+@pytest.fixture(autouse=True)
+def set_plotting_lib():
+    vp.set_option("plotting_lib", "matplotlib")
```

Comment:
you can switch back to default by adding a yeild statement
```
yeild
vp.set_option("plotting_lib", "plotly")
```

Reply:
Thanks added.